.changeset/menu.md

```diff
@@ -0,0 +1,23 @@
+---
+'@leafygreen-ui/menu': major
+---
+
+[LG-4121](https://jira.mongodb.org/browse/LG-4121): Replaces `usePortal` prop with `renderMode` prop. `renderMode="inline"` and `renderMode="portal"` are deprecated, and all popover elements should migrate to using the top layer. The old default was `usePortal = true`, and the new default is `renderMode = 'top-layer'`. 
```

Comment:
small nit: `renderMode="inline"` has no spaces but `usePortal = true` and `renderMode = 'top-layer'` do